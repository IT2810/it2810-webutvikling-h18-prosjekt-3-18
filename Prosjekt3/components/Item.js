import React, { Component,  } from 'react';
import { View, Text, Button, TouchableOpacity, StyleSheet} from 'react-native';
import { MaterialIcons } from '@expo/vector-icons';

class Item extends Component {
    constructor(props) {
        super(props);
        this.onPressDelete = this.onPressDelete.bind(this);
    }

    render() {
        return (
            <View style={styles.items}>
                {this.props.children}
                <TouchableOpacity title="Delete" onPress={this.onPressDelete}>
                    <MaterialIcons name="delete" size={40} color="black" />
                </TouchableOpacity>
            </View>
        );
    }

    onPressDelete() {
        this.props.onDelete(this.props.id);
    }

}
const styles = StyleSheet.create({
        items: {
            borderTopWidth: 1,
            flexDirection: 'row',
            borderStyle: 'dashed'
        }


});
export default Item;
