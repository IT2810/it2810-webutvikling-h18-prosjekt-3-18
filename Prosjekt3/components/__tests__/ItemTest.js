import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import Item from '../../components/Item';

it('renders correctly', () => {
    const tree = renderer.create(<Item>Snapshot test!</Item>).toJSON();
    expect(tree).toMatchSnapshot();
});
