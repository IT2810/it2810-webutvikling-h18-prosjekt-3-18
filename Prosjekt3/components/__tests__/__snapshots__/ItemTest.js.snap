// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "borderTopWidth": 1,
      "flexDirection": "row",
    }
  }
>
  Snapshot test!
  <View
    accessibilityComponentType="button"
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Delete
      </Text>
    </View>
  </View>
</View>
`;
